File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/MessageDbEntityMapper.java
package uk.co.victoriajanedavis.chatapp.data.mappers;

import io.reactivex.annotations.NonNull;
import uk.co.victoriajanedavis.chatapp.data.model.db.MessageDbModel;
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper;
import uk.co.victoriajanedavis.chatapp.domain.entities.MessageEntity;

public class MessageDbEntityMapper extends Mapper<MessageDbModel, MessageEntity> {

    @Override
    public MessageEntity mapFrom(MessageDbModel from) {
        MessageEntity entity = new MessageEntity();

        entity.setUuid(from.getUuid());
        entity.setChatUuid(from.getChatUuid());
        entity.setUserUuid(from.getUserUuid());
        entity.setUserUsername(from.getUserUsername());
        entity.setText(from.getText());
        entity.setCreated(from.getCreated());
        entity.setFromCurrentUser(from.isFromCurrentUser());

        return entity;
    }
}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/MessageDbEntityMapper.kt
package uk.co.victoriajanedavis.chatapp.data.mappers

import uk.co.victoriajanedavis.chatapp.data.model.db.MessageDbModel
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper
import uk.co.victoriajanedavis.chatapp.domain.entities.MessageEntity

class MessageDbEntityMapper : Mapper<MessageDbModel, MessageEntity>() {

    override fun mapFrom(from: MessageDbModel): MessageEntity {
        return MessageEntity(
            uuid = from.uuid,
            chatUuid = from.chatUuid,
            userUuid = from.userUuid,
            userUsername = from.userUsername,
            text = from.text,
            created = from.created,
            isFromCurrentUser = from.isFromCurrentUser
        )
    }
}
